curl/tests/mock_curl.cc

```diff
@@ -0,0 +1,361 @@
+#define BUILDING_LIBCURL
+#include <cstring>
+#include <stdarg.h>
+#include "mock_curl.hpp"
+#include <iostream>
+
+using namespace std;
+
+/*
+ * libcurl implementations below. We are mocking necessary methods
+ * of the CURL API to ensure that our wrapper is making the calls
+ * to libcurl that we expect.
+ */
+
+/*
+ * Error struct implementations below... <for testing various libcurl failures>
+ * These are for the pre-client inits. The rest of the exceptions can use an
+ * enum inside of the curl_impl class
+ */
+error_mode test_failure_mode = success;
```

Comment:
This should be declared `static`. It doesn't need to be exposed externally, that's what the methods are for.